internal/typeinfo/member.go

```diff
@@ -13,6 +13,7 @@ var scannerInterface = reflect.TypeOf((*sql.Scanner)(nil)).Elem()
 type ValueLocator interface {
 	ArgType() reflect.Type
 	String() string
+	IDString() string
```

Comment:
[Nit]: Can we change the name to `ID`? The `String` bit is already part of the type signature.

Reply:
And can we state in a comment here or in the implementers that the invariant is that this function **must** return a unique representation which lets us differentiate between the different objects?

Reply:
Name changed to ID, I was 50/50 on which to choose.

I want to be careful around language here, we don't actually want a unique ID for each ValueLocator. In a way that is the problem that caused the bug in the first place since we were differentiating between map keys that looked the same but were different objects. We want to check weather the textual representation of the members looks the same.

I have added a comment to try and capture what the ID represents.

Reply:
I see what you mean. To me ID was the identifier for the underlying concept, not the id for each concrete instantiation to be different from each other. Maybe there is a better name like `repr` in Python or `hash` or something like that.

Reply:
The problem with the name and comment now is that it is hard to know why there is a need for a new function when we have `String` already.

Reply:
I've changed the `String` function to `Desc` meaning description and the `ID` function to `Repr` for representation. Hopefully that clears it up a bit.